proposal/color-4-new-spaces-js.d.ts.md

```diff
@@ -127,61 +127,107 @@ get space(): KnownColorSpace;
 
 #### `channels`
 
-Returns an array of channel values for [`internal`], with missing channels
-converted to `0`.
+Returns an array of channel values (excluding `alpha`) for [`internal`], with
+[missing channels][missing components] converted to `0`.
+
+[missing components]: ./color-4-new-spaces.md#missing-components
 
 ```ts
-get channels(): string | number;
+get channels(): [number, number, number];
 ```
 
 #### `channelsOrNull`
 
-Returns an array of channel values for [`internal`], with missing channels
-converted to `null`.
+Returns an array of channel values (excluding `alpha`) for [`internal`], with
+[missing channels][missing components] converted to `null`.
 
 ```ts
-get channelsOrNull(): ChannelValue;
+get channelsOrNull(): [ChannelValue, ChannelValue, ChannelValue];
 ```
 
 #### `isLegacy`
 
-Returns whether [`internal`] is in a legacy color space (`rgb`, `hsl`, or
+Returns whether [`internal`] is in a [legacy color space] (`rgb`, `hsl`, or
 `hwb`).
 
 ```ts
 get isLegacy(): boolean;
 ```
 
+[legacy color space]: ./color-4-new-spaces.md#legacy-color
+
 #### `isInGamut`
 
-Returns whether [`internal`] is in-gamut for its color space (as opposed to
+Returns whether [`internal`] is [in-gamut] for its color space (as opposed to
 having one or more of its channels out of bounds, like `rgb(300 0 0)`).
 
 ```ts
 get isInGamut(): boolean;
 ```
 
+[in-gamut]: ./color-4-new-spaces.md#coloris-in-gamut
+
 #### `channel`
 
-Returns the value of the given `channel` in [`internal`], after converting it to
-`space` if necessary. It should be used instead of the old channel-specific
-functions such as `color.red()` and `color.hue()`.
+Returns the value of the given `channel` in [`internal`]. If `space` is set, it
+will first [convert] to that `space`.
```

Comment:
This, and some of the members below, are still written as documentation rather than a specification. As a rule of thumb, constructs like "it will" don't belong in a specification—they exist to communicate to users what is going to happen when the function is invoked, but we want to communicate to implementers what _must_ happen when the function is invoked (very precisely with an explicit ordering).

If it helps, try writing it as a bulleted list of steps rather than just a paragraph.

Reply:
> I'm considering having the specification for `channel` simply refer to the spec for [color.channel](https://github.com/sass/sass/blob/main/proposal/color-4-new-spaces.md#colorchannel-1). However, that handles alpha- should `channel` here also allow for the alpha component? 

Yeah, I think any API that takes a channel name should be able to handle alpha as well, even if there are also explicit alpha APIs (because every color has an alpha).

> The alpha question also applies to `isChannelMissing` -> `color.is-missing` . I could rename `isChannelMissing` to `isMissing` and then refer to the spec for [color.is-missing](https://github.com/sass/sass/blob/main/proposal/color-4-new-spaces.md#coloris-missing-1) 

I think the name `isChannelMissing` is clearer here, but I think referring to the Sass function is fine.

Reply:
I've updated these. My main concern was conflating channels and components, but I think it's matching the Sass Color 4 proposal fairly closely now.